src/coprocessor/dag/executor/aggregation.rs

```diff
@@ -81,6 +79,7 @@ pub struct HashAggExecutor {
     related_cols_offset: Vec<usize>, // offset of related columns
     src: Box<Executor>,
     count: i64,
+    first_collect: bool,
```

Comment:
what is `first_collect` used for?

Reply:
Since we may collect statistics more than once in `streaming`, while for each executor should be considered only one time. @breeswish

Reply:
I think it's a wrong place to increase `executor_count` in `collect_metrics_into`, at least it is misleading. There is no relationship between `first_collect`, `collect_metrics_into` and `executor_counter`.

I have two solutions:

1. I suggest to remove `first_collect` and clear the `metrics.executor_count` after every flush, just like `metrics.(others)` so that this function will have a singular semantic: whenever calling this function, it will generate a set of metrics (although `executor_count` will be the same each time logically).

2. If you really would like to eliminate duplicate counting each time in streaming, I suggest to add a standalone function, named like `count_children()`, and call it only once at the most outside function, which collects the children count into a metrics.

Reply:
I prefer solution 1.

Reply:
It may not work as expected since not all executor has a metrics.